### PR TITLE
Improved auth

### PR DESCRIPTION
**Pull request by orestkozolup**
Changed the way of authorization status check. Adjusted header behaviour to the new method. Removed a file with the old check of authorization.

### CODE REVIEW

File: client/src/App.vue [hunk 1]
```diff
@@ -31,7 +30,10 @@ export default {
   },
 
   created () {
-    if (Authorized.isAuthorized()) {
+    if (
+      window.localStorage.getItem('token') &&
+      window.localStorage.getItem('token') !== ''
```
Comment (igorkokotko) [inline client/src/App.vue]: Why you remove isAuthorized? And why you need extra check for window.localStorage.getItem('token') !== ''?

File: client/src/layouts/Header.vue [hunk 2]
```diff
@@ -89,13 +91,34 @@ export default {
...
+    ...mapGetters({
+      token: 'auth/getToken'
+    })
+  },
+
+  async created () {
+    try {
+      const auth = await isAuthorized()
+      this.isAuthenticated = auth
+    } catch (e) {
```
Comment (igorkokotko) [inline client/src/layouts/Header.vue:82]: Catch should not be an empty block.

**Review by igorkokotko — CHANGES_REQUESTED** [review]

File: client/src/views/Authentication/AuthEmailAction.vue [hunk 3]
```diff
@@ -95,10 +95,18 @@ export default {
   },
 
   beforeRouteEnter: (to, from, next) => {
-    if (Authorized.isAuthorized()) {
-      return next('/feed')
-    }
-    next()
+    isAuthorized()
+      .then(res => {
+        if (res === true) {
```
Comment (igorkokotko) [inline client/src/views/Authentication/AuthEmailAction.vue]: return res ? next('/feed') : next() would be shorter and you can remove finally block.

File: client/src/views/Authentication/AuthLogin.vue [hunk 4]
```diff
@@ -10,17 +10,25 @@
...
     AuthLoginComponent
   },
   beforeRouteEnter: (to, from, next) => {
-    if (Authorized.isAuthorized()) {
-      return next('/feed')
-    }
-    next()
+    isAuthorized()
+      .then(res => {
+        if (res === true) {
```
Comment (igorkokotko) [inline client/src/views/Authentication/AuthLogin.vue]: Please look comment above

**Review by igorkokotko — CHANGES_REQUESTED** [review]

File: client/src/services/Authorized.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { checkToken } from '@/services/auth.js'
+
+export const isAuthorized = function () {
+  return new Promise((resolve, reject) => {
+    checkToken()
+      .then((res) => {
+        if (res.data.id === window.localStorage.getItem('profileId')) {
```
Comment (igorkokotko) [inline client/src/services/Authorized.js]: res.data.id === window.localStorage.getItem('profileId') if this statement equal to true then you could resolve this result

File: client/src/views/Settings/index.vue [hunk 6]
```diff
@@ -6,18 +6,29 @@
...
+          return next('/login')
+        }
+        if (to.path === '/settings') {
+          return next('/settings/editProfile')
+        }
+      })
+      .catch(() => {
+        return next('/login')
+      })
+      .finally(() => {
```
Comment (igorkokotko) [inline client/src/views/Settings/index.vue]: Why you need finally?

**Review by igorkokotko — CHANGES_REQUESTED** [review]

**Review by de5peratelighter — APPROVED** [review]

**Review by ghost — COMMENTED** [review]
Molodec

**Review by ghost — APPROVED** [review]

**Review by igorkokotko — APPROVED** [review]

**Pull request merged**